cmake/tools.cmake

```diff
@@ -90,22 +91,36 @@ endif ()
 
 if (LINKER_NAME)
     if (COMPILER_CLANG)
-        find_program (LLD_PATH NAMES ${LINKER_NAME})
-        if (NOT LLD_PATH)
-            message (FATAL_ERROR "Using linker ${LINKER_NAME} but can't find its path.")
-        endif ()
-
-        # This a temporary quirk to emit .debug_aranges with ThinLTO
-        set (LLD_WRAPPER "${CMAKE_CURRENT_BINARY_DIR}/ld.lld")
-        configure_file ("${CMAKE_CURRENT_SOURCE_DIR}/cmake/ld.lld.in" "${LLD_WRAPPER}" @ONLY)
+        if (LINKER_NAME MATCHES "lld")
+            find_program (LLD_PATH NAMES ${LINKER_NAME})
+            if (NOT LLD_PATH)
+                message (FATAL_ERROR "Using linker ${LINKER_NAME} but can't find its path.")
+            endif ()
 
-        set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} --ld-path=${LLD_WRAPPER}")
-        set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} --ld-path=${LLD_WRAPPER}")
+            if (CMAKE_BUILD_TYPE MATCHES "Rel")
```

Comment:
Maybe it is better to spell `Release` and `RelWithDebInfo` like in other places, for better `grep`-ability.
Also you should use `CMAKE_BUILD_TYPE_UC`  instead, otherwise it is case sensitive match.